unity_jobs_simulator/Assets/Scripts/Robot_Controller.cs

```diff
@@ -0,0 +1,273 @@
+        {
+            targetPos = new List<float>() {0, 0, 0, 0, -90, 0};
+        }
+        if(Input.GetKeyDown(KeyCode.Keypad6) || Input.GetKeyDown(KeyCode.Alpha6))
+        {
+            targetPos = new List<float>() {0, 0, 0, 0, 0, 90};
+        }
+    }
+
+    void FixedUpdate()
+    {
+        updateCurrentDrivePos(); // We need to update the current position of the Robot to ensure we know exactly where it is
+        
+        if(samePos(_currentDrivePos, targetPos)) // If we are at the preset target, FixedUpdate shouldn't run so we return nothing
+        {
+            // checkAngleErrors();
+            return;
+        }
+
+        if(!(samePos(_currentDrivePos, _enroute))) // If the route has changed, we need to restart the motion
+        {
+            _enroute = new List<float>(targetPos);
+            _incAngles = new List<float>(_zeroList);
+        }
+
+        if(samePos(_incAngles, _zeroList)) // If we don't have increments for the angles, then we aren't going anywhere and need to initialize these increments
+        {
+            getIncrements(); // This calculates the values for increments given the target angles, the current positions, and the maximum velocity
+            setIncAngles(); // This calculates the incremental angles given the increments and the target angles
+            _enroute = new List<float>(targetPos); // We set our enroute position as this is where our robot movement will be moving to
+            _nextAngles = addLists(_currentDrivePos, _incAngles); // This increments the angles by one increment
+            _t1 = DateTime.Now; // Time.realtimeSinceStartup; // We record the start time for this motion
+        }
+
+        if(samePos(_nextAngles, targetPos)) // If the next Angle is the final target position, then this is the final increment
+        {
+            adjustRobotAngles(targetPos, 0f); // We set the final angle to the final target, and we set the joint the velocities to zero
+            _incAngles = new List<float>(_zeroList); // We've reached the final location so we do not want to increment the angles anymore so we set this to zero
+            _t2 = DateTime.Now; // Time.realtimeSinceStartup; // We record the final time 
+            TimeSpan timeDiff = _t2.Subtract(_t1);
+            Debug.Log("Time took for that motion was: " + timeDiff.ToString()); // Log the time taken
+            return;
+        }
+        else // Otherwise, the robot is incrementing the angles to the next incremental position
+        {
+            adjustRobotAngles(_nextAngles, _maxVel); // We adjust the robot to the next position
+            _nextAngles = addLists(_nextAngles, _incAngles); // We increment the nextAngle list up by the incremental angles
+        }
+    }
+
+    public void updateCurrentDrivePos()
+    {
+        _currentDrivePos = new List<float>(); // Creates a new list to store the current Drive target positions in
```

Comment:
This variable should be renamed to _currentDriveTarget since it's not actually giving position. Why is this useful?